baked_in.go

```diff
@@ -1404,6 +1407,74 @@ func isFile(fl FieldLevel) bool {
 	panic(fmt.Sprintf("Bad field type %T", field.Interface()))
 }
 
+func isImage(fl FieldLevel) bool {
+	mimetypes := map[string]bool{
+		"image/bmp": true,
+		"image/cis-cod": true,
+		"image/gif": true,
+		"image/ief": true,
+		"image/jpeg": true, 
+		"image/jp2": true, 
+		"image/jpx": true, 
+		"image/jpm": true, 
+		"image/pipeg": true,
+		"image/png": true, 
+		"image/svg+xml": true,
+		"image/tiff": true,
+		"image/webp": true,
+		"image/x-cmu-raster": true,
+		"image/x-cmx": true,
+		"image/x-icon": true,
+		"image/x-portable-anymap": true,
+		"image/x-portable-bitmap": true,
+		"image/x-portable-graymap": true,
+		"image/x-portable-pixmap": true,
+		"image/x-rgb": true,
+		"image/x-xbitmap": true,
+		"image/x-xpixmap": true,
+		"image/x-xwindowdump": true,
+	}
+	field := fl.Field()
+
+	switch field.Kind() {
+	case reflect.String:
+		filePath := field.String()
+		fileInfo, err := os.Stat(filePath)
+
+		if err != nil {
+			return false
+		}
+
+		if fileInfo.IsDir() {
+			return false
+		}
+
+		file, err := os.OpenFile(filePath, os.O_RDWR, 0644)
+		if err != nil {
+			return false
+		}
+		defer file.Close()
+
+		mime, err := mimetype.DetectReader(file)
+		if err != nil {
+			return false
+		}
+
+		_, err = file.Seek(0, io.SeekStart)
+		if err != nil {
+			panic(fmt.Sprintf("Could not reset seek start for file. Error: %s", err))
+		}
```

Comment:
I'm not sure why the seek is required? should be removed from what I can tell.

Reply:
While reading data on the mime type of the file, the offset for the reader is moved away from the beginning of the file and this may cause issues for other parts of the program that need to read from that file. Using seek ensures that after the mime type is read, the offset of the reader is returned to the beginning of the file so that there will be no issues for other parts of the program that will need to read from the file.

Reply:
Right I understand what `Seek` does and why one would need to use it but in this case the file handle is closed when this function ends because of the `defer file.Close()` and nothing else tries to read using that handle and so there is no need to Seek from what I can tell.

Reply:
Right, got you.